clang-tools-extra/clang-tidy/bugprone/AssertSideEffectCheck.cpp

```diff
@@ -41,6 +41,10 @@ AST_MATCHER_P2(Expr, hasSideEffect, bool, CheckFunctionCalls,
   }
 
   if (const auto *OpCallExpr = dyn_cast<CXXOperatorCallExpr>(E)) {
+    if (const auto *FuncDecl = OpCallExpr->getDirectCallee())
+      if (const auto *MethodDecl = dyn_cast<CXXMethodDecl>(FuncDecl))
+        return !MethodDecl->isConst();
```

Comment:
should it be `if (MethodDecl->isConst()) return false`; otherwise code in line 49 could be a dead code.

Reply:
this makes we wonder if we need the comparisons agains the different operator kinds at all, or if we should treat all operators the same, based on their constness.

Reply:
Reporting all non const operator calls would not be feasible, because e.g. the `->` operator of `std::optional` is not const but is side-effect free.